### PR TITLE
Split regex and extend tests

### PR DESCRIPTION
**Pull request by fregante**
Splitting the regex makes it much easier to edit.

If this isn't well-liked, I propose one of:

- [ ] Have a build (even manual) so that you can edit this version but ship a regex literal
- [ ] Use snapshots so that the full readable regex appears somewhere on the repo

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -123,14 +128,15 @@ test('main #2', t => {
 		'foo/-bar#123',
 		'foo/-#123',
 		'foo/.bar#123',
+		'foo/bar.#123',
 		'foo/...#123',
-		'#123',
+		'#999',
```
Comment (fregante) [inline test.js:133]: Side note: The tests are really hard to parse when one is missing, the whole "diff" breaks and it's not really clear what's - and what's +.

This is why I changes a couple to avoid the repetitive `123`

I'll look for an alternative later maybe.

Edit: done: https://github.com/sindresorhus/issue-regex/pull/19

File: index.js [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+const {source: organization} = /(?<organization>[a-z\d](?:[a-z\d-]{0,37}[a-z\d])?)/;
+const {source: repository} = /(?<repository>[\w.-]{1,100})/;
+const {source: reservedRepository} = /(?<!\/\.{1,2})/;
+const {source: issueNumber} = /(?<issueNumber>[1-9]\d{0,9})/;
+const {source: initialDelimiter} = /(?<!\w)/;
+
 export default function issueRegex() {
-	return /(?<!\w)(?:(?<organization>[a-z\d](?:[a-z\d-]{0,37}[a-z\d])?)\/(?<repository>[\w.-]{1,100}))?(?<!(?:\/\.{1,2}))#(?<issueNumber>[1-9]\d{0,9})\b/gi;
+	return new RegExp(
+		`${initialDelimiter}(?:${organization}\\/${repository})?${reservedRepository}#${issueNumber}\\b`,
```
Comment (sindresorhus) [inline index.js]: Nitpick, but this could be a top-level string variable.

**Comment by sindresorhus** [comment]
This looks fine to me.

**Comment by fregante** [comment]
By the way I may have found some bugs while reworking the tests, so if you do your Node 18 / dependency update commit I’ll follow up with the test updates. #16 would have to wait.

**Comment by sindresorhus** [comment]
> so if you do your Node 18

Are we planning any other breaking changes? If not, I don't see the strong need to do a major update just for Node.js 18.

**Comment by fregante** [comment]
Just this: https://github.com/xojs/xo/releases/tag/v0.57.0

But the solution is to stay on XO 0.56
